packages/create-amplify/src/initial_project_file_generator.ts

```diff
@@ -42,7 +38,7 @@ export class InitialProjectFileGenerator {
       JSON.stringify(packageJsonContent, null, 2)
     );
 
-    if (process.env.PACKAGE_MANAGER_EXECUTABLE === 'yarn-modern') {
+    if (this.packageManager.name === 'yarn-modern') {
```

Comment:
now that we have YarnClassic/Moder initializer - can we move this if and if below as implementation detail there?

Reply:
Unfortunately we have to generate the lock file after `amplify/` folder created, which happens after the initializer runs.

Reply:
In that case I'd inject `PackageManagerController` into `InitialProjectFileGenerator` and add `initializeAmplifyFolder` on `PackageManagerController` - some subtypes will implement this, others will have it non-op. 
(this should also take care of that `if` below which installs dependency in `amplify` folder if `packageManagerController.installDependencies` can take optional parameter `pathRelativeToProjectRoot` or so.


Another thought after looking more into this is:
the last line can turn into `packageManagerController.executeCommand('tsc', tscArgs)` (or so), so that `binaryRunner` can become internal detail of package manager abstraction.

Reply:
There are other ways to handle this besides this file knowing about yarn-specific things. Maybe PackageManagerController needs an `initializeFiles` method of it's own that can be passed into this class. For npm and pnpm it's a noop but for yarn it does this stuff.

Reply:
added `initializeAmplifyFolder`

Didn't add `executeCommand` because it looks unnecessary at this moment